### PR TITLE
Client: organize store files/folders

### PR DESCRIPTION
**Pull request by belcurv**
This PR proposes that we:

1.  collect all of the client's store-related files/folders within a single `/store` folder, rather than have separate `/src`-level folders for actions and reducers.

2.  move redux init/config code out of `index.js` and into its own `/store/store.js`.

The `<Header>` component is updated to the new `/store/Actions/authentication` path.

### CODE REVIEW

**Comment by belcurv** [comment]
Oh - I also added the constant `GET_USER_STATUS` to the authentication action. Making actions constants and exporting them reduces typos and enables IDE auto-complete.

```js
export const GET_USER_STATUS = 'GET_USER_STATUS';

export function getUser() {
  return async (dispatch) => {
    const response = await axios.get('/api/profile');
    dispatch({
      type: GET_USER_STATUS,
      payload: response.data,
    });
  };
}
```

File: client/src/Components/Header/Header.jsx [hunk 1]
```diff
@@ -7,14 +7,18 @@ import './Header.css';
 // connect to redux and get action creators
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
-import getUser from '../../Actions/authentiation';
+import { getUser } from '../../Actions/authentiation';
+
 // reads from store
 const mapStateToProps = state => state;
```
Comment (terrancexin) [inline client/src/Components/Header/Header.jsx:13]: you can delete this line if you don't need to access any redux props

File: client/src/Components/Header/Header.jsx [hunk 2]
```diff
@@ -50,41 +54,42 @@ class Header extends React.Component {
...
 
 Header.propTypes = {
   user: PropTypes.objectOf(PropTypes.any),
-  getUser: PropTypes.func,
+  actions: PropTypes.shape({
+    getUser: PropTypes.func,
+  }),
 };
 
 export default connect(mapStateToProps, mapDispatchToProps)(Header);
```
Comment (terrancexin) [inline client/src/Components/Header/Header.jsx:95]: and `mapStateToProps` can just be `null` here

File: client/src/Components/Header/Header.jsx (hunk 1 above)
Comment (belcurv) [reply client/src/Components/Header/Header.jsx:13]: This is from @Dereje1's work on auth setup, as is the subsequent mapStateToProps below.

Comment (belcurv) [reply client/src/Components/Header/Header.jsx:13]: We do use `this.props.user.authenticated` in the setReady method though.

**Review by eolculnamo2 — APPROVED** [review]

**Pull request merged**